### PR TITLE
feature/java preview

### PR DESCRIPTION
**Pull request by baywet**
changes in this pull request:
- adds randomization of java compile folder (to avoid conflicts in case of parrallels execution)
- adds runsettings  configuration files (with parallelization of work for java)
- adds defensive programming for the runsettings parser to avoid taking into account default/empty values
- adds the ability for the java test runner to run tests against a preview version of the SDK
- adds pipelines to run tests against a preview version of the SDK (using resources and steps that build the SDK first)

### CODE REVIEW

File: TestsCommon/RunSettings.cs [hunk 1]
```diff
@@ -29,27 +29,32 @@ public RunSettings(TestParameters parameters)
...
+                    "OBJECTIVEC" => Languages.ObjC,
+                    "OBJECTIVE-C" => Languages.ObjC,
+                    _ => Languages.CSharp
+                };
+
+            if (!string.IsNullOrEmpty(dllPath) && !dllPath.Contains(dashdash))
             {
-                throw new ArgumentException("File specified with DllPath in Test.runsettings doesn't exist!");
+                DllPath = dllPath;
+                if ((Language == Languages.CSharp || Language == Languages.Java) && !File.Exists(dllPath) && !Directory.Exists(dllPath)) // java uses a directory
```
Comment (zengin) [inline TestsCommon/RunSettings.cs:51]: Please use a different parameter for Java. Let's keep `dllPath` for Microsoft.Graph.dll.

File: JavaV1KnownFailureTests/Test.runsettings [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<RunSettings>
+  <RunConfiguration>
+    <MaxCpuCount>4</MaxCpuCount>
+  </RunConfiguration>
+  <TestRunParameters>
+    <Parameter name="Version" value="---VersionPlaceholder---" />
+    <Parameter name="DllPath" value="C:/sources/github" />
+    <Parameter name="KnownFailuresRequested" value="---KnownFailuresRequestedPlaceholder---" />
+    <Parameter name="Language" value="---LanguagePlaceHolder---" />
```
Comment (zengin) [inline JavaV1KnownFailureTests/Test.runsettings:10]: Don't we know these values beforehand for these specific projects, i.e. Version = V1, KnownFailuresRequested=true, Language=Java.

Comment (zengin) [reply JavaV1KnownFailureTests/Test.runsettings:10]: Maybe we can hardcode these and skip the transform steps.

**Review by zengin — CHANGES_REQUESTED** [review]

File: JavaV1KnownFailureTests/Test.runsettings (hunk 2 above)
Comment (baywet) [reply JavaV1KnownFailureTests/Test.runsettings:10]: my idea here was that I didn't want to create another 2 test projects for the preview version. So any java project can be used to test against a preview version.

File: TestsCommon/RunSettings.cs (hunk 1 above)
Comment (baywet) [reply TestsCommon/RunSettings.cs]: fixed

**Comment by baywet** [comment]
@zengin this is ready for final review. I requeued the existing pipelines to double check I didn't break anything. And here is a run with the Java preview one. (with correct branches for resources)
https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=31836&view=results

**Review by zengin — APPROVED** [review]

**Pull request merged**